spec/lib/danger/ci_sources/bitrise_spec.rb

```diff
@@ -0,0 +1,64 @@
+require "danger/ci_source/xcode_server"
+
+describe Danger::XcodeServer do
+  let(:valid_env) do
+    {
+      "BITRISE_PULL_REQUEST" => "4",
+      "BITRISE_IO" => "true",
+      "GIT_REPOSITORY_URL" => "git@github.com:artsy/eigen"
+    }
+  end
+
+  let(:invalid_env) do
+    {
+      "CIRCLE" => "true"
+    }
+  end
+
+  let(:source) { described_class.new(valid_env) }
+
+  describe ".validates_as_pr?" do
+    it "validates when the required env variables are set" do
+      expect(described_class.validates_as_pr?(valid_env)).to be true
+    end
+
+    it "does not validate when the required env variables are not set" do
+      expect(described_class.validates_as_pr?(invalid_env)).to be false
+    end
+  end
+
+  describe ".validates_as_ci?" do
+    it "validates when the required env variables are set" do
+      expect(described_class.validates_as_ci?(valid_env)).to be true
+    end
```

Comment:
Typically nice to include a case where you set only the variables required for `validates_as_ci?` to be valid, but that would not be the case for `validates_as_pr?`. For example see this [example](https://github.com/danger/danger/blob/master/spec/lib/danger/ci_sources/travis_spec.rb#L25-L35) in the Travis CI tests